src/test/mir-opt/issues/issue_59352.num_to_digit.PreCodegen.after.mir

```diff
@@ -19,13 +19,6 @@ fn num_to_digit(_1: char) -> u32 {
             let mut _9: isize;           // in scope 2 at $DIR/issue-59352.rs:14:8: 14:23
         }
     }
-    scope 3 (inlined #[track_caller] Option::<u32>::unwrap) { // at $DIR/issue-59352.rs:14:26: 14:50
```

Comment:
`Option::unwrap` not MIR-inlining anymore suggests its body is more complex (maybe not enough to stop LLVM from inlining itself, but at least making it more work to simplify).

If a lot of `panic!` users in the standard library suddenly increased in complexity, this could *easily* explain the regressions, on their own (though without alleviating them, I can't be sure there aren't other causes too).

In general, if something is causing non-trivial changes in MIR tests, a PR shouldn't be just `--bless`'d and merged, but an additional review from @rust-lang/wg-mir-opt (or anyone else in @rust-lang/compiler with MIR experience) should be required.

Even if some of these MIR optimizations are not on by default, they're good canaries for potential issues down the line.

Reply:
Checking now what happens when we implement `unwrap()` with `panic()` (the function) instead of `panic!()` (the macro).

Reply:
https://github.com/rust-lang/rust/pull/92140